src/pages/SearchPage.js

```diff
@@ -1,37 +1,38 @@
 import React from 'react';
-import movies from '../mocks/movie';
+import {connect} from 'react-redux';
+
+import {startMoviesRequest} from '../actions/movies';
+import moviesSelector from '../selectors/movies';
 import SearchHeader from '../components/SearchHeader';
 import ResultHeader from '../components/ResultHeader';
 import MovieList from '../components/MovieList';
 import Footer from '../components/Footer';
 
-export default class SearchPage extends React.PureComponent {
-
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      sortBy: 'release_date'
-    }
-  }
-
-  onSortChange = (sortBy) => {
-    this.setState({sortBy: sortBy});
-  }
-
+export class SearchPage extends React.PureComponent {
   render() {
+    const {movies} = this.props;
     // Uncomment this line to test ErrorBoundary component
     // throw new Error('Something went wrong');
     return (
       <div className="search-page">
         <SearchHeader />
-        <ResultHeader resultNumber={movies.length}
-                      sortBy={this.state.sortBy}
-                      onChange={this.onSortChange} />
-        <MovieList movies={movies} 
-                   onPageChange={this.props.onPageChange}/>
+        {!!movies.length && (
+          <React.Fragment>
+          <ResultHeader resultNumber={movies.length} />
+          <MovieList movies={movies} 
+                handleMovieSelected={this.props.handleMovieSelected}/>
+          </React.Fragment>
+        )}
         <Footer />
       </div>
     )
   }
 }
+
+const mapStateToProps = ({movies, filters: {sortBy}}) => {
+  return {
+    movies: moviesSelector(movies, sortBy)
```

Comment:
selectors === <3

But one thing that you must pay attention is that selectors are made to hide the complexity of your state to your components/containers. Your selector should be auto-sufficient to know how to threat your state.

would be better to just sent your state to your selector and your selector will retrieve your movies and sortBy from the state, if you decide to change your state, you need to just update your selector not your component also.

Reply:
So you are saying I should keep track of the search query and the search by filter in the state, right?  You mentioned it was a bad practice to have the state of the forms in redux, but I guess you meant the bad practice is updating it every time the user presses a key. But what you are saying is the final values of search query and search by filter should be in the state, right?